drivers/hmis/app/graphql/mutations/join_household.rb

```diff
@@ -0,0 +1,103 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  class JoinHousehold < BaseMutation
+    argument :receiving_household_id, ID, required: true
+    argument :joining_enrollment_inputs, [Types::HmisSchema::EnrollmentRelationshipInput], required: true
+
+    field :receiving_household, Types::HmisSchema::Household, null: false
+    field :donor_household, Types::HmisSchema::Household, null: true # Will be null if there are no remaining members
+    # todo @martha - possible that there will be multiple households with same ID? mutation should require project_id?
+    def resolve(receiving_household_id:, joining_enrollment_inputs:)
+      receiving_enrollments = Hmis::Hud::Enrollment.
+        where(household_id: receiving_household_id).
+        viewable_by(current_user).
+        includes(:project, :household)
+      access_denied! if receiving_enrollments.empty?
+
+      receiving_household = receiving_enrollments.first.household
+
+      receiving_project = receiving_enrollments.map(&:project).uniq.sole
+      access_denied! unless current_permission?(permission: :can_split_households, entity: receiving_project)
+
+      # The mutation input is a list, so convert it to a hashmap for ease of access
+      map_enrollment_id_to_relationship = joining_enrollment_inputs.map { |e| [e.enrollment_id, e.relationship_to_hoh] }.to_h
+
+      joining_enrollment_ids = map_enrollment_id_to_relationship.keys
+      joining_enrollments = Hmis::Hud::Enrollment.
+        viewable_by(current_user).
+        where(id: joining_enrollment_ids).
+        includes(:household)
+      access_denied! unless joining_enrollments.count == joining_enrollment_inputs.count
+
+      raise 'Cannot merge enrollments from another project' unless joining_enrollments.pluck(:project_pk).uniq.sole == receiving_project.id
+
+      donor_household = joining_enrollments.map(&:household).uniq.sole
+      remaining_enrollments = Hmis::Hud::Enrollment.
+        viewable_by(current_user). # Restrict to this data source
+        where(household_id: donor_household.household_id).
+        where.not(id: joining_enrollment_ids)
```

Comment:
I believe this can be written as:
`remaining_enrollments  = donor_household.enrollments - joining_enrollments`
(it will be an array, not a scope)